### ISSUE TITLE
Price Source Exchange error for pure market making (Cryptocom)

### ISSUE THREAD

**Issue by leastchaos** [CoinAlpha/hummingbot#4659]
**Describe the bug** 
unable to run PMM with cryptocom as external price source

**Steps To Reproduce**
// A concise description of the **steps to reproduce the buggy behavior**:
Run bot with external market as price source and crypto com as exchange
Refer to config file

**Screenshots**
```
2021-11-06 04:41:32,974 - 210118 - hummingbot.client.hummingbot_application - ERROR - Unknown error during initialization.
Traceback (most recent call last):
  File "/root/hummingbot-1/hummingbot/strategy/pure_market_making/start.py", line 66, in start
    ext_market = create_paper_trade_market(price_source_exchange, [asset_trading_pair])
  File "/root/hummingbot-1/hummingbot/connector/exchange/paper_trade/__init__.py", line 27, in create_paper_trade_market
    obt_obj = obt_class(**obt_kwargs)
TypeError: __init__() missing 1 required positional argument: 'shared_client'
2021-11-06 04:41:32,977 - 210118 - hummingbot.client.hummingbot_application - INFO - start command initiated.
2021-11-06 04:41:33,157 - 210118 - hummingbot.connector.exchange.ascend_ex.ascend_ex_exchange - INFO - Network status has changed to NetworkStatus.CONNECTED. Starting networking...
2021-11-06 04:41:33,273 - 210118 - hummingbot.connector.exchange.ascend_ex.ascend_ex_order_book_tracker - INFO - Initialized order book for BTC-USDT. 1/1 completed.
2021-11-06 04:41:33,276 - 210118 - hummingbot.connector.exchange.ascend_ex.ascend_ex_api_order_book_data_source - INFO - Subscribed to ['BTC-USDT'] orderbook trading and delta streams...
2021-11-06 04:41:37,168 - 210118 - hummingbot.client.hummingbot_application - INFO - stop command initiated.
2021-11-06 04:41:37,169 - 210118 - hummingbot.core.utils.async_utils - ERROR - Unhandled error in background task: list.remove(x): x not in list
Traceback (most recent call last):
  File "/root/hummingbot-1/hummingbot/core/utils/async_utils.py", line 9, in safe_wrapper
    return await c
  File "/root/hummingbot-1/hummingbot/client/command/stop_command.py", line 37, in stop_loop
    self.clock.remove_iterator(self.strategy)
  File "hummingbot/core/clock.pyx", line 83, in hummingbot.core.clock.Clock.remove_iterator
    self._child_iterators.remove(iterator)
ValueError: list.remove(x): x not in list
```

**Release version**
0.45.0

**Attachments**
// Please attach your config file and log file to help us debug. It would be difficult for us to help you without those! 
log file: entire log refer above
config: [conf_pure_mm_2.txt](https://github.com/CoinAlpha/hummingbot/files/7489903/conf_pure_mm_2.txt)

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**

**Comment by BannAgainVA** [comment]
Top is my discord and YoutacVA27 or anything with vedarius a russells pocture on it is my accounts and do not release any information to these scammers. Thats why they can't access them 318-3**-8050 and 816-446-8063 is my mobile numbers. This is Vedarius A Russell . Lawsuits coming for anyone who tampers with my accounts

**Comment by gerald-hb** [comment]
I was able to replicate the same issue.
Docker build 0.45
```2021-11-06 06:43:27,102 - 1 - hummingbot.client.hummingbot_application - ERROR - Unknown error during initialization.
Traceback (most recent call last):
  File "/home/hummingbot/hummingbot/strategy/pure_market_making/start.py", line 66, in start
    ext_market = create_paper_trade_market(price_source_exchange, [asset_trading_pair])
  File "/home/hummingbot/hummingbot/connector/exchange/paper_trade/__init__.py", line 27, in create_paper_trade_market
    obt_obj = obt_class(**obt_kwargs)
TypeError: __init__() missing 1 required positional argument: 'shared_client'
2021-11-06 06:43:27,118 - 1 - hummingbot.client.hummingbot_application - INFO - start command initiated.
2021-11-06 06:43:27,691 - 1 - hummingbot.connector.exchange.kucoin.kucoin_exchange - INFO - Network status has changed to NetworkStatus.CONNECTED. Starting networking...
2021-11-06 06:43:27,993 - 1 - hummingbot.connector.exchange.kucoin.kucoin_api_user_stream_data_source - INFO - Connecting to wss://ws-api.kucoin.com/endpoint?token=2neAiuYvAU737TOajb2U3uT8AEZqSWYe0fBD4LoHuXJDSC7gIzJiH4kNTWhCPISWo6nDpAe7aUaD5tJbxAFv1Dkz56AFfKah3r6P-AxCXhuUEdvRquETqXwZIngtMdMrjqPnP-biofGK1iYo0qDMJ1WnFAuRV4B7JBvJHl5Vs9Y=.VrH6zP1dMONHGuiAnWAdeQ==&acceptUserMessage=true.
2021-11-06 06:43:29,873 - 1 - hummingbot.connector.exchange.kucoin.kucoin_order_book_tracker - INFO - Initialized order book for BTC-USDT. 1/1 completed.
2021-11-06 06:45:26,963 - 1 - hummingbot.client.hummingbot_application - INFO - stop command initiated.
2021-11-06 06:45:26,963 - 1 - hummingbot.core.utils.async_utils - ERROR - Unhandled error in background task: list.remove(x): x not in list
Traceback (most recent call last):
  File "/home/hummingbot/hummingbot/core/utils/async_utils.py", line 9, in safe_wrapper
    return await c
  File "/home/hummingbot/hummingbot/client/command/stop_command.py", line 37, in stop_loop
    self.clock.remove_iterator(self.strategy)
  File "hummingbot/core/clock.pyx", line 83, in hummingbot.core.clock.Clock.remove_iterator
    self._child_iterators.remove(iterator)
ValueError: list.remove(x): x not in list
2021-11-06 06:46:05,673 - 1 - hummingbot.client.hummingbot_application - INFO - stop command initiated.
2021-11-06 06:46:05,673 - 1 - hummingbot.core.utils.async_utils - ERROR - Unhandled error in background task: list.remove(x): x not in list
Traceback (most recent call last):
  File "/home/hummingbot/hummingbot/core/utils/async_utils.py", line 9, in safe_wrapper
    return await c
  File "/home/hummingbot/hummingbot/client/command/stop_command.py", line 37, in stop_loop
    self.clock.remove_iterator(self.strategy)
  File "hummingbot/core/clock.pyx", line 83, in hummingbot.core.clock.Clock.remove_iterator
    self._child_iterators.remove(iterator)
ValueError: list.remove(x): x not in list
2021-11-06 06:46:14,841 - 1 - hummingbot.client.hummingbot_application - INFO - stop command initiated.
2021-11-06 06:46:14,841 - 1 - hummingbot.core.utils.async_utils - ERROR - Unhandled error in background task: list.remove(x): x not in list```
[logs_conf_pure_mm_4.log](https://github.com/CoinAlpha/hummingbot/files/7490001/logs_conf_pure_mm_4.log)
[conf_4pmm.txt](https://github.com/CoinAlpha/hummingbot/files/7490004/conf_4pmm.txt)

**Comment by sisco0** [comment]
@leastchaos @gerald-hb Could you kindly test if the current implementation at my branch fixes the problem? As I do not have API keys configured yet.

New branch for testing:
https://github.com/sisco0/hummingbot/tree/fix-crypto_com-order-book-tracker

The changes introduced specified a default parameter to the `shared_client` as pointed out in the commit diff https://github.com/sisco0/hummingbot/commit/d21c61ecd016513637e456dadbeef2fb5c2f4ed5.

**Comment by dennisocana** [comment]
Thanks to @sisco0 for contributing a fix for this issue to be included in version 0.46 release.
